src/modules/common/components/Headroom.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import raf from 'raf'
+
+import style from './Headroom.css'
+
+class Headroom extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.any.isRequired,
+    scrollHeight: PropTypes.number.isRequired,
+    pinStart: PropTypes.number.isRequired,
+    stickyAncestor: PropTypes.node,
+    height: PropTypes.number
+  }
+
+  static defaultProps = {
+    pinStart: 0
+  }
+
+  constructor (props) {
+    super(props)
+    this.handleEvent = this.handleEvent.bind(this)
+    this.update = this.update.bind(this)
+    this.ticking = false
+    this.lastScrollY = 0
+    this.lastTransform = 0
+    this.lastKnownScrollY = 0
+    this.currentScrollY = 0
+    this.direction = 'down'
+    this.state = {transform: 0}
+  }
+
+  componentDidMount () {
+    window.addEventListener('scroll', this.handleEvent)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('scroll touchstart touchmove', this.handleEvent)
+  }
+
+  getScrollY = () => {
+    if (window.pageYOffset !== undefined) {
+      return window.pageYOffset
+    } else if (window.scrollTop !== undefined) {
+      return window.scrollTop
+    } else {
+      return (document.documentElement || document.body.parentNode || document.body).scrollTop
+    }
+  }
+
+  truncate (transform, currentScrollY) {
```

Comment:
documentation would be nice :D

Reply:
Yup, documentation and testing will be added. I created the pull request, so that you can check the header on your devices. I want to see how you like it.